File: Algorithms (Programmers)/Camouflage.js
/* 2021.04.22. 2021 Programmers High score kit
 * Camouflage
 */

function solution(clothes) {
    let answer = 1;
    const map = new Map();
    for(let [dress,kind] of clothes){
        if(map.get(kind)){
            map.set(kind,map.get(kind)+1)
        }else{
        map.set(kind,1);
        }
    }
    for(let kind of map.keys()){
        console.log(kind)
        answer *= map.get(kind)+1;
    }

    return answer-1;
}

console.log(solution([["yellowhat", "headgear"], ["bluesunglasses", "eyewear"], ["green_turban", "headgear"]]))

File: programmers/Camouflage.py
# 2021.04.19. 2021 Programmers High score kit
# Camouflage

from collections import defaultdict
def solution(clothes):
    # 1. Making dress table
    dress = defaultdict(list)
    for name,kind in clothes:
        dress[kind].append(name)
    # 2. Multiply each kind+1.
    answer = 1
    for kind in dress:
        answer *= len(dress[kind]) +1
    # 3. Except nothing wear.
    return answer-1
